# ImageMagick6 f4454af87ec1 calls an undefined ThrowFPXrException; the macro it defines is ThrowFPXException
--- ImageMagick-6.9.13-59/coders/fpx.c.orig
+++ ImageMagick-6.9.13-59/coders/fpx.c
@@ -886,7 +886,7 @@
   memory_limit=20000000;
   fpx_status=FPX_SetToolkitMemoryLimit(&memory_limit);
   if (fpx_status != FPX_OK)
-    ThrowFPXrException(DelegateError,"UnableToInitializeFPXLibrary");
+    ThrowFPXException(DelegateError,"UnableToInitializeFPXLibrary");
   tile_width=64;
   tile_height=64;
   colorspace.numberOfComponents=3;
@@ -930,7 +930,7 @@
       fpx_status=FPX_SetJPEGCompression(flashpix,(unsigned short)
         image->quality == UndefinedCompressionQuality ? 75 : image->quality);
       if (fpx_status != FPX_OK)
-        ThrowFPXrException(DelegateError,"UnableToSetJPEGLevel");
+        ThrowFPXException(DelegateError,"UnableToSetJPEGLevel");
     }
   /*
     Set image summary info.
@@ -952,7 +952,7 @@
         summary_info.title.ptr=(unsigned char *) AcquireQuantumMemory(
           length+MaxTextExtent,sizeof(*summary_info.title.ptr));
       if (summary_info.title.ptr == (unsigned char *) NULL)
-        ThrowFPXrException(DelegateError,"UnableToSetImageTitle");
+        ThrowFPXException(DelegateError,"UnableToSetImageTitle");
       (void) CopyMagickString((char *) summary_info.title.ptr,label,
         MaxTextExtent);
     }
@@ -968,13 +968,13 @@
     }
   fpx_status=FPX_SetSummaryInformation(flashpix,&summary_info);
   if (fpx_status != FPX_OK)
-    ThrowFPXrException(DelegateError,"UnableToSetSummaryInfo");
+    ThrowFPXException(DelegateError,"UnableToSetSummaryInfo");
   /*
     Initialize FlashPix image description.
   */
   quantum_info=AcquireQuantumInfo(image_info,image);
   if (quantum_info == (QuantumInfo *) NULL)
-    ThrowFPXrException(ResourceLimitError,"MemoryAllocationFailed");
+    ThrowFPXException(ResourceLimitError,"MemoryAllocationFailed");
   pixels=GetQuantumPixels(quantum_info);
   fpx_info.numberOfComponents=(unsigned int) colorspace.numberOfComponents;
   for (i=0; i < (ssize_t) fpx_info.numberOfComponents; i++)
@@ -1113,37 +1113,37 @@
         {
           fpx_status=FPX_SetImageAffineMatrix(flashpix,&affine);
           if (fpx_status != FPX_OK)
-            ThrowFPXrException(DelegateError,"UnableToSetAffineMatrix");
+            ThrowFPXException(DelegateError,"UnableToSetAffineMatrix");
         }
       if (aspect_ratio_valid != MagickFalse)
         {
           fpx_status=FPX_SetImageResultAspectRatio(flashpix,&aspect_ratio);
           if (fpx_status != FPX_OK)
-            ThrowFPXrException(DelegateError,"UnableToSetAspectRatio");
+            ThrowFPXException(DelegateError,"UnableToSetAspectRatio");
         }
       if (color_twist_valid != MagickFalse)
         {
           fpx_status=FPX_SetImageColorTwistMatrix(flashpix,&color_twist);
           if (fpx_status != FPX_OK)
-            ThrowFPXrException(DelegateError,"UnableToSetColorTwist");
+            ThrowFPXException(DelegateError,"UnableToSetColorTwist");
         }
       if (contrast_valid != MagickFalse)
         {
           fpx_status=FPX_SetImageContrastAdjustment(flashpix,&contrast);
           if (fpx_status != FPX_OK)
-            ThrowFPXrException(DelegateError,"UnableToSetContrast");
+            ThrowFPXException(DelegateError,"UnableToSetContrast");
         }
       if (sharpen_valid != MagickFalse)
         {
           fpx_status=FPX_SetImageFilteringValue(flashpix,&sharpen);
           if (fpx_status != FPX_OK)
-            ThrowFPXrException(DelegateError,"UnableToSetFilteringValue");
+            ThrowFPXException(DelegateError,"UnableToSetFilteringValue");
         }
       if (view_rect_valid != MagickFalse)
         {
           fpx_status=FPX_SetImageROI(flashpix,&view_rect);
           if (fpx_status != FPX_OK)
-            ThrowFPXrException(DelegateError,"UnableToSetRegionOfInterest");
+            ThrowFPXException(DelegateError,"UnableToSetRegionOfInterest");
         }
     }
   (void) FPX_CloseImage(flashpix);
